addons/mail/static/src/core/common/partner_suggestion_compare_fn.js

```diff
@@ -0,0 +1,141 @@
+/* @odoo-module */
+
+import { cleanTerm } from "@mail/utils/common/format";
+import { registry } from "@web/core/registry";
+
+/**
+ * Registry of functions to sort partner suggestions.
+ * The expected value is an object of the following form:
+ *    { sequence: number, fn: (partner1, partner2, { searchTerms: string, thread?: Thread }) => number|undefined }
+ * Note that the compare functions will be call in the context of the
+ * caller (i.e. the SuggestionService).
+ */
+export const partnerSuggestionCompareRegistry = registry.category(
```

Comment:
Regarding name for later, I'm wondering whether we would want to use this registry for other sorting purpose (new DM, invite member, etc)?

Reply:
We could name it `partnerCompareRegistry` and be more precise if required in the future